addons/mail/static/src/models/thread/thread.js

```diff
@@ -1230,6 +1230,17 @@ function factory(dependencies) {
             this._currentPartnerLastNotifiedIsTyping = isTyping;
         }
 
+        /**
+         * Handles change of messages on the current thread. In particular, all
+         * non-main caches currently displaying this thread should check for the
+         * newly added (or removed) messages according to their domain.
+         */
+        _onMessagesChanged() {
+            for (const threadView of this.threadViews) {
+                threadView.fetchNonMainCacheMessages();
+            }
+        }
+
```

Comment:
Explanation is unclear, I don't understand what this does.
Code is short, so would guess it should be straight-forward at least with the code... but considering the fetch method isn't actually a fetch, I'm quite lost...